### PR TITLE
chore(release): publish agoric-upgrade-13

### PR DESCRIPTION
**Pull request by mhofman**
## Packages that have NEWS.md updates

```diff
--- a/golang/cosmos/CHANGELOG.md
+++ b/golang/cosmos/CHANGELOG.md
@@ -3,6 +3,30 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## [0.35.0-u13.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/cosmos@0.35.0-u12.0...@agoric/cosmos@0.35.0-u13.0) (2023-12-07)
+
+
+### Features
+
+* **cosmos:** un-wire x/crisis ([#8582](https://github.com/Agoric/agoric-sdk/issues/8582)) ([19404a3](https://github.com/Agoric/agoric-sdk/commit/19404a3c5cd10d9c454f190b60cbf0aa715f605c))
+* **x/swingset:** auto-provision smart wallet ([5073800](https://github.com/Agoric/agoric-sdk/commit/5073800f87c92a194a21c9cc59ad1bb570a39b95))
+* **x/swingset:** refuse smart wallet messages if not provisioned ([50a5a55](https://github.com/Agoric/agoric-sdk/commit/50a5a557956562fc05b8c2defd99a0102a35153f))
+* pick up return-grants from latest cosmos-sdk ([975533e](https://github.com/Agoric/agoric-sdk/commit/975533e088f96c80cf8d6f6b7c5f45665e3495ba))
+* update ibc-go to v4, adapt packages and API ([6672e5c](https://github.com/Agoric/agoric-sdk/commit/6672e5cb780c8baa530ac59bff27d793e1ff8f50))
+
+
+### Bug Fixes
+
+* handle hang-on-halt behavior from agoric-labs/cosmos-sdk[#305](https://github.com/Agoric/agoric-sdk/issues/305) ([7ae9a70](https://github.com/Agoric/agoric-sdk/commit/7ae9a705478727d1d7ace7c665861cfee14b5f28))
+* update dependencies to fix tests ([9d750dd](https://github.com/Agoric/agoric-sdk/commit/9d750dd39993d380a5e889ea7faa4bb78c3257aa))
+
+
+### Reverts
+
+* Revert "fix: handle hang-on-halt behavior from agoric-labs/cosmos-sdk#305" ([3a37f5e](https://github.com/Agoric/agoric-sdk/commit/3a37f5e472405d1d3bee2075b47ecd3632ba4073)), closes [agoric-labs/cosmos-sdk#305](https://github.com/agoric-labs/cosmos-sdk/issues/305) [agoric-labs/cosmos-sdk#305](https://github.com/agoric-labs/cosmos-sdk/issues/305)
+
+
+
 ## [0.35.0-u12.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/cosmos@0.35.0-u11wf.0...@agoric/cosmos@0.35.0-u12.0) (2023-11-10)
 
 
--- a/packages/SwingSet/CHANGELOG.md
+++ b/packages/SwingSet/CHANGELOG.md
@@ -3,6 +3,15 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+### [0.32.3-u13.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/swingset-vat@0.32.3-u12.0...@agoric/swingset-vat@0.32.3-u13.0) (2023-12-07)
+
+
+### Bug Fixes
+
+* handle Node 18 breaking change ([#8600](https://github.com/Agoric/agoric-sdk/issues/8600)) ([2c64440](https://github.com/Agoric/agoric-sdk/commit/2c64440d5b5957076ad9ac7b9243de3b600c859f))
+
+
+
 ### [0.32.3-u12.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/swingset-vat@0.32.3-u11wf.0...@agoric/swingset-vat@0.32.3-u12.0) (2023-11-10)
 
 
--- a/packages/agoric-cli/CHANGELOG.md
+++ b/packages/agoric-cli/CHANGELOG.md
@@ -3,6 +3,32 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## [0.22.0-u13.0](https://github.com/Agoric/agoric-sdk/compare/agoric@0.21.2-u12.0...agoric@0.22.0-u13.0) (2023-12-07)
+
+
+### ⚠ BREAKING CHANGES
+
+* restrict exports from 'agoric' package
+
+### Features
+
+* **cosmos:** un-wire x/crisis ([#8582](https://github.com/Agoric/agoric-sdk/issues/8582)) ([19404a3](https://github.com/Agoric/agoric-sdk/commit/19404a3c5cd10d9c454f190b60cbf0aa715f605c))
+
+
+### Bug Fixes
+
+* **agops:** fix continuing id lookup in oracle setPrice ([0ac9697](https://github.com/Agoric/agoric-sdk/commit/0ac969746c7337d45287fd040b636d27d296a062))
+* **agoric:** print errors while following ([deab7e4](https://github.com/Agoric/agoric-sdk/commit/deab7e4065fe5dd6cc1a3c5bf8c0db861cb7856b))
+* **casting:** dont crash on bad capdata ([3f01369](https://github.com/Agoric/agoric-sdk/commit/3f0136994c0b4033c872f9593cabb19bab05f01c))
+* **cli:** handle not found error in vstorage requests ([8111c43](https://github.com/Agoric/agoric-sdk/commit/8111c4344bcaa6e06085020813e2431cba9d1632))
+
+
+### Miscellaneous Chores
+
+* restrict exports from 'agoric' package ([63c1abb](https://github.com/Agoric/agoric-sdk/commit/63c1abbfcb0710bdd0a22382eb4a683216a6c453))
+
+
+
 ### [0.21.2-u12.0](https://github.com/Agoric/agoric-sdk/compare/agoric@0.21.2-u11wf.0...agoric@0.21.2-u12.0) (2023-11-10)
 
 **Note:** Version bump only for package agoric
--- a/packages/casting/CHANGELOG.md
+++ b/packages/casting/CHANGELOG.md
@@ -3,6 +3,16 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+### [0.4.3-u13.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/casting@0.4.3-u12.0...@agoric/casting@0.4.3-u13.0) (2023-12-07)
+
+
+### Bug Fixes
+
+* **casting:** dont crash on bad capdata ([3f01369](https://github.com/Agoric/agoric-sdk/commit/3f0136994c0b4033c872f9593cabb19bab05f01c))
+* **casting:** properly follow an unpopulated state entry ([48296f7](https://github.com/Agoric/agoric-sdk/commit/48296f78f783e8e5ae1e5fbe92ae10e5e7880837))
+
+
+
 ### [0.4.3-u12.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/casting@0.4.3-u11wf.0...@agoric/casting@0.4.3-u12.0) (2023-11-10)
 
 **Note:** Version bump only for package @agoric/casting
--- a/packages/cosmic-swingset/CHANGELOG.md
+++ b/packages/cosmic-swingset/CHANGELOG.md
@@ -3,6 +3,20 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## [0.42.0-u13.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/cosmic-swingset@0.42.0-u12.0...@agoric/cosmic-swingset@0.42.0-u13.0) (2023-12-07)
+
+
+### Features
+
+* add exporter.getHostKV() API ([16435d2](https://github.com/Agoric/agoric-sdk/commit/16435d20e9ede86916a54c7bae54ecfc59e4c950)), closes [#8523](https://github.com/Agoric/agoric-sdk/issues/8523)
+
+
+### Bug Fixes
+
+* export state-sync snapshot without a DB write-lock ([2d388c2](https://github.com/Agoric/agoric-sdk/commit/2d388c2d51425b9d98b79d54ea6a2de96da67042)), closes [#8523](https://github.com/Agoric/agoric-sdk/issues/8523)
+
+
+
 ## [0.42.0-u12.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/cosmic-swingset@0.42.0-u11wf.0...@agoric/cosmic-swingset@0.42.0-u12.0) (2023-11-10)
 
 
--- a/packages/inter-protocol/CHANGELOG.md
+++ b/packages/inter-protocol/CHANGELOG.md
@@ -3,6 +3,15 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+### [0.16.2-u13.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/inter-protocol@0.16.2-u12.0...@agoric/inter-protocol@0.16.2-u13.0) (2023-12-07)
+
+
+### Bug Fixes
+
+* **inter-protocol:** brands in CLI need not start with A-Z ([#8258](https://github.com/Agoric/agoric-sdk/issues/8258)) ([fd9a44d](https://github.com/Agoric/agoric-sdk/commit/fd9a44db84e2cb9c9c78f850be7dbb8911650d54))
+
+
+
 ### [0.16.2-u12.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/inter-protocol@0.16.2-u11wf.0...@agoric/inter-protocol@0.16.2-u12.0) (2023-11-10)
 
 **Note:** Version bump only for package @agoric/inter-protocol
--- a/packages/solo/CHANGELOG.md
+++ b/packages/solo/CHANGELOG.md
@@ -3,6 +3,15 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+### [0.10.4-u13.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/solo@0.10.4-u12.0...@agoric/solo@0.10.4-u13.0) (2023-12-07)
+
+
+### Bug Fixes
+
+* **solo:** properly expose `/wallet/bridge.html` ([#8533](https://github.com/Agoric/agoric-sdk/issues/8533)) ([0095b0d](https://github.com/Agoric/agoric-sdk/commit/0095b0d8ef4cfd42e17b64f391c30eac8a5f018a))
+
+
+
 ### [0.10.4-u12.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/solo@0.10.4-u11wf.0...@agoric/solo@0.10.4-u12.0) (2023-11-10)
 
 
--- a/packages/swing-store/CHANGELOG.md
+++ b/packages/swing-store/CHANGELOG.md
@@ -3,6 +3,15 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+### [0.9.2-u13.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/swing-store@0.9.2-u12.0...@agoric/swing-store@0.9.2-u13.0) (2023-12-07)
+
+
+### Features
+
+* add exporter.getHostKV() API ([16435d2](https://github.com/Agoric/agoric-sdk/commit/16435d20e9ede86916a54c7bae54ecfc59e4c950)), closes [#8523](https://github.com/Agoric/agoric-sdk/issues/8523)
+
+
+
 ### [0.9.2-u12.0](https://github.com/Agoric/agoric-sdk/compare/@agoric/swing-store@0.9.2-u11wf.0...@agoric/swing-store@0.9.2-u12.0) (2023-11-10)
 
 
```

### CODE REVIEW

**Comment by mhofman** [comment]
The release commit is https://github.com/Agoric/agoric-sdk/pull/8633/commits/be59032e2f86fa1becfbbdd4a1f32561ad71ccc6, rest comes from previous `dev-upgrade-13` branch PRs.

File: packages/deployment/upgrade-test/Readme.md [hunk 1]
```diff
@@ -24,7 +14,7 @@ By default pre-releases use the lastest image tagged `dev` in our [container rep
 a specific build:
 
 ```shell
-DEST_IMAGE=docker pull ghcr.io/agoric/agoric-sdk:20230515033839-e56ae7
+DEST_IMAGE=ghcr.io/agoric/agoric-sdk:20230515033839-e56ae7
```
Comment (toliaqat) [inline packages/deployment/upgrade-test/Readme.md:17]: Do we need to change this tag?

File: golang/cosmos/app/app.go [hunk 2]
```diff
@@ -825,8 +825,8 @@ func NewAgoricApp(
 	return app
 }
 
-// upgrade12Handler performs standard upgrade actions plus custom actions for upgrade-12.
-func upgrade12Handler(app *GaiaApp, targetUpgrade string) func(sdk.Context, upgradetypes.Plan, module.VersionMap) (module.VersionMap, error) {
+// upgrade13Handler performs standard upgrade actions plus custom actions for upgrade-13.
+func upgrade13Handler(app *GaiaApp, targetUpgrade string) func(sdk.Context, upgradetypes.Plan, module.VersionMap) (module.VersionMap, error) {
```
Comment (toliaqat) [inline golang/cosmos/app/app.go:828]: If there is nothing specific in this function, should we rename it to just upgradeHandler?

File: packages/deployment/upgrade-test/Makefile [hunk 3]
```diff
@@ -43,4 +43,4 @@ run_test:
 shell:
 	docker exec -it `docker ps --latest --format json | jq -r .Names` bash
 
-.PHONY: local_sdk agoric-upgrade-7-2 agoric-upgrade-8 agoric-upgrade-8-1 agoric-upgrade-9 agoric-upgrade-10 agoric-upgrade-11 agoric-upgrade-12 build build_test run
+.PHONY: local_sdk agoric-upgrade-13 build build_test run
```
Comment (toliaqat) [inline packages/deployment/upgrade-test/Makefile:41]: Why we not need to run all the upgrades like before?

**Review by toliaqat — COMMENTED** [review]
Looks good overall. 
Left couple of comments and wanted you to double if we need to update the following file?

`packages/xsnap/package.json`

https://github.com/Agoric/agoric-sdk/blob/prepare-release-20231207T020132/packages/xsnap/package.json#L31

File: packages/deployment/upgrade-test/Readme.md (hunk 1 above)
Comment (mhofman) [reply packages/deployment/upgrade-test/Readme.md:17]: It's just an example in the readme

File: golang/cosmos/app/app.go (hunk 2 above)
Comment (mhofman) [reply golang/cosmos/app/app.go:828]: That's how we've been managing upgrade handlers. This change is coming from #8588 which prepared the upgrade-13 branch

File: packages/deployment/upgrade-test/Makefile (hunk 3 above)
Comment (mhofman) [reply packages/deployment/upgrade-test/Makefile:41]: Because those are now part of agoric-3-proposals. This was some cleanup done in #8588

**Review by toliaqat — APPROVED** [review]
@mhofman I am conditionally approving this PR. Please check if you need to update version numbers in the following file before merging this PR.

https://github.com/Agoric/agoric-sdk/blob/prepare-release-20231207T020132/packages/xsnap/package.json

**Comment by mhofman** [comment]
> @mhofman I am conditionally approving this PR. Please check if you need to update version numbers in the following file before merging this PR.
> 
> https://github.com/Agoric/agoric-sdk/blob/prepare-release-20231207T020132/packages/xsnap/package.json

Oh I see what you mean. The xsnap-lockdown version didn't get bumped, and arguably it should to represent the fact the bundle on chain is now different. I'll see what I can do.

**Comment by mhofman** [comment]
@toliaqat I forced a version bump on `xsnap-lockdown`, triggering a bump on `xsnap` as well.
I also included the fix from #8527 (which is still pending review...) since the tag tool is currently broken

Overall diff since reviewed: [be59032..5a6cdeb](https://github.com/Agoric/agoric-sdk/compare/be59032e2f86fa1becfbbdd4a1f32561ad71ccc6..5a6cdeb0c18ae9700d706445acf402f8d1e873c3)

PTAL

**Comment by toliaqat** [comment]
@mhofman thank you.

I think because of the broken tool, we missed u11wf.0 tag updates in the past as well. 
Can you check if we need to update them for @agoric/assert and @agoric/access-token pacakges.
https://github.com/Agoric/agoric-sdk/blob/prepare-release-20231207T020132/packages/agoric-cli/package.json#L37

And update version in eslint-config
https://github.com/Agoric/agoric-sdk/blob/prepare-release-20231207T020132/packages/eslint-config/package.json#L3

**Comment by mhofman** [comment]
> I think because of the broken tool, we missed u11wf.0 tag updates in the past as well.

To be clear, it's totally normal for packages that didn't experience any changes to not get their version bumped. In the case of `xsnap-lockdown` it actually was a problem because we patched `ses` resulting in a different content of the distributed bundle. I don't think it was your intent to point this specific issue out, but your review made me realize the problem, which I've addressed for that package specifically.

The other packages can stay without a version change.

**Review by toliaqat — APPROVED** [review]

**Comment by toliaqat** [comment]
🚀

**Pull request merged**